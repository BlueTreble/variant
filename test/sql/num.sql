\set ECHO 0
\set QUIET 1


\set baseline_type numeric
-- Space separated string
\set test_types 'int2 int4 int8 float real numeric'
-- Used in array[ :base_values ]
\set base_values '-1, 0, 1, NULL' 
\set compare_value 0


BEGIN;
\i test/helpers/tap_setup.sql
\i test/helpers/common.sql

CREATE TEMP VIEW test_type AS
  SELECT * FROM unnest( string_to_array(
      :'test_types'
      , ' '
    ) ) AS u(test_type)
;

CREATE TEMP VIEW base_value AS
  SELECT base_value:::baseline_type FROM unnest(
    array[ :base_values ]
  ) AS bv(base_value)
;

CREATE TEMP VIEW compare_value AS
  SELECT :'compare_value':::baseline_type AS compare_value
;

\i test/helpers/type_setup.sql

SELECT plan(sum(test_count)::int) FROM plan;

\i test/helpers/type.sql


-- vi: expandtab sw=2 ts=2
