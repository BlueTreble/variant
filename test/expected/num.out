\set ECHO none


ok 1..0


1..320
ok 1 - No NULL variants
ok 2 - Verify result of cast '-1.1'::numeric(2,1)
ok 3 - Verify result of cast '0.1'::numeric(2,1)
ok 4 - Verify result of cast '1.1'::numeric(2,1)
ok 5 - Verify result of cast NULL::numeric(2,1)
ok 6 - Verify result of cast '-1.1'::numeric
ok 7 - Verify result of cast '0.1'::numeric
ok 8 - Verify result of cast '1.1'::numeric
ok 9 - Verify result of cast NULL::numeric
ok 10 - SET base_variant_b
ok 11 - Verify text_in(text_out()) works for ("numeric(2,1)",-1.1)
ok 12 - Verify text_in(text_out()) works for ("numeric(2,1)",0.1)
ok 13 - Verify text_in(text_out()) works for ("numeric(2,1)",1.1)
ok 14 - Verify text_in(text_out()) works for ("numeric(2,1)",)
ok 15 - Verify text_in(text_out()) works for (numeric,-1.1)
ok 16 - Verify text_in(text_out()) works for (numeric,0.1)
ok 17 - Verify text_in(text_out()) works for (numeric,1.1)
ok 18 - Verify text_in(text_out()) works for (numeric,)
ok 19
ok 20
ok 21
ok 22
ok 23
ok 24
ok 25
ok 26
ok 27 - No NULL variants
ok 28 - Verify result of cast '0.1'::numeric(2,1)
ok 29 - Verify result of cast '0.1'::numeric
ok 30 - SET compare_variant_b
ok 31 - Verify text_in(text_out()) works for ("numeric(2,1)",0.1)
ok 32 - Verify text_in(text_out()) works for (numeric,0.1)
ok 33 - check casted vs uncasted for '0.1'::numeric <   '-1.1'::numeric
ok 34 - check casted vs uncasted for '0.1'::numeric <   '0.1'::numeric
ok 35 - check casted vs uncasted for '0.1'::numeric <   '1.1'::numeric
ok 36 - check casted vs uncasted for '0.1'::numeric <   NULL::numeric
ok 37 - check casted vs uncasted for '0.1'::numeric <   '-1.1'::numeric(2,1)
ok 38 - check casted vs uncasted for '0.1'::numeric <   '0.1'::numeric(2,1)
ok 39 - check casted vs uncasted for '0.1'::numeric <   '1.1'::numeric(2,1)
ok 40 - check casted vs uncasted for '0.1'::numeric <   NULL::numeric(2,1)
ok 41 - check casted vs uncasted for '0.1'::numeric(2,1) <   '-1.1'::numeric
ok 42 - check casted vs uncasted for '0.1'::numeric(2,1) <   '0.1'::numeric
ok 43 - check casted vs uncasted for '0.1'::numeric(2,1) <   '1.1'::numeric
ok 44 - check casted vs uncasted for '0.1'::numeric(2,1) <   NULL::numeric
ok 45 - check casted vs uncasted for '0.1'::numeric(2,1) <   '-1.1'::numeric(2,1)
ok 46 - check casted vs uncasted for '0.1'::numeric(2,1) <   '0.1'::numeric(2,1)
ok 47 - check casted vs uncasted for '0.1'::numeric(2,1) <   '1.1'::numeric(2,1)
ok 48 - check casted vs uncasted for '0.1'::numeric(2,1) <   NULL::numeric(2,1)
ok 49 - check casted vs uncasted for '0.1'::numeric <=  '-1.1'::numeric
ok 50 - check casted vs uncasted for '0.1'::numeric <=  '0.1'::numeric
ok 51 - check casted vs uncasted for '0.1'::numeric <=  '1.1'::numeric
ok 52 - check casted vs uncasted for '0.1'::numeric <=  NULL::numeric
ok 53 - check casted vs uncasted for '0.1'::numeric <=  '-1.1'::numeric(2,1)
ok 54 - check casted vs uncasted for '0.1'::numeric <=  '0.1'::numeric(2,1)
ok 55 - check casted vs uncasted for '0.1'::numeric <=  '1.1'::numeric(2,1)
ok 56 - check casted vs uncasted for '0.1'::numeric <=  NULL::numeric(2,1)
ok 57 - check casted vs uncasted for '0.1'::numeric(2,1) <=  '-1.1'::numeric
ok 58 - check casted vs uncasted for '0.1'::numeric(2,1) <=  '0.1'::numeric
ok 59 - check casted vs uncasted for '0.1'::numeric(2,1) <=  '1.1'::numeric
ok 60 - check casted vs uncasted for '0.1'::numeric(2,1) <=  NULL::numeric
ok 61 - check casted vs uncasted for '0.1'::numeric(2,1) <=  '-1.1'::numeric(2,1)
ok 62 - check casted vs uncasted for '0.1'::numeric(2,1) <=  '0.1'::numeric(2,1)
ok 63 - check casted vs uncasted for '0.1'::numeric(2,1) <=  '1.1'::numeric(2,1)
ok 64 - check casted vs uncasted for '0.1'::numeric(2,1) <=  NULL::numeric(2,1)
ok 65 - check casted vs uncasted for '0.1'::numeric  =  '-1.1'::numeric
ok 66 - check casted vs uncasted for '0.1'::numeric  =  '0.1'::numeric
ok 67 - check casted vs uncasted for '0.1'::numeric  =  '1.1'::numeric
ok 68 - check casted vs uncasted for '0.1'::numeric  =  NULL::numeric
ok 69 - check casted vs uncasted for '0.1'::numeric  =  '-1.1'::numeric(2,1)
ok 70 - check casted vs uncasted for '0.1'::numeric  =  '0.1'::numeric(2,1)
ok 71 - check casted vs uncasted for '0.1'::numeric  =  '1.1'::numeric(2,1)
ok 72 - check casted vs uncasted for '0.1'::numeric  =  NULL::numeric(2,1)
ok 73 - check casted vs uncasted for '0.1'::numeric(2,1)  =  '-1.1'::numeric
ok 74 - check casted vs uncasted for '0.1'::numeric(2,1)  =  '0.1'::numeric
ok 75 - check casted vs uncasted for '0.1'::numeric(2,1)  =  '1.1'::numeric
ok 76 - check casted vs uncasted for '0.1'::numeric(2,1)  =  NULL::numeric
ok 77 - check casted vs uncasted for '0.1'::numeric(2,1)  =  '-1.1'::numeric(2,1)
ok 78 - check casted vs uncasted for '0.1'::numeric(2,1)  =  '0.1'::numeric(2,1)
ok 79 - check casted vs uncasted for '0.1'::numeric(2,1)  =  '1.1'::numeric(2,1)
ok 80 - check casted vs uncasted for '0.1'::numeric(2,1)  =  NULL::numeric(2,1)
ok 81 - check casted vs uncasted for '0.1'::numeric  >= '-1.1'::numeric
ok 82 - check casted vs uncasted for '0.1'::numeric  >= '0.1'::numeric
ok 83 - check casted vs uncasted for '0.1'::numeric  >= '1.1'::numeric
ok 84 - check casted vs uncasted for '0.1'::numeric  >= NULL::numeric
ok 85 - check casted vs uncasted for '0.1'::numeric  >= '-1.1'::numeric(2,1)
ok 86 - check casted vs uncasted for '0.1'::numeric  >= '0.1'::numeric(2,1)
ok 87 - check casted vs uncasted for '0.1'::numeric  >= '1.1'::numeric(2,1)
ok 88 - check casted vs uncasted for '0.1'::numeric  >= NULL::numeric(2,1)
ok 89 - check casted vs uncasted for '0.1'::numeric(2,1)  >= '-1.1'::numeric
ok 90 - check casted vs uncasted for '0.1'::numeric(2,1)  >= '0.1'::numeric
ok 91 - check casted vs uncasted for '0.1'::numeric(2,1)  >= '1.1'::numeric
ok 92 - check casted vs uncasted for '0.1'::numeric(2,1)  >= NULL::numeric
ok 93 - check casted vs uncasted for '0.1'::numeric(2,1)  >= '-1.1'::numeric(2,1)
ok 94 - check casted vs uncasted for '0.1'::numeric(2,1)  >= '0.1'::numeric(2,1)
ok 95 - check casted vs uncasted for '0.1'::numeric(2,1)  >= '1.1'::numeric(2,1)
ok 96 - check casted vs uncasted for '0.1'::numeric(2,1)  >= NULL::numeric(2,1)
ok 97 - check casted vs uncasted for '0.1'::numeric   > '-1.1'::numeric
ok 98 - check casted vs uncasted for '0.1'::numeric   > '0.1'::numeric
ok 99 - check casted vs uncasted for '0.1'::numeric   > '1.1'::numeric
ok 100 - check casted vs uncasted for '0.1'::numeric   > NULL::numeric
ok 101 - check casted vs uncasted for '0.1'::numeric   > '-1.1'::numeric(2,1)
ok 102 - check casted vs uncasted for '0.1'::numeric   > '0.1'::numeric(2,1)
ok 103 - check casted vs uncasted for '0.1'::numeric   > '1.1'::numeric(2,1)
ok 104 - check casted vs uncasted for '0.1'::numeric   > NULL::numeric(2,1)
ok 105 - check casted vs uncasted for '0.1'::numeric(2,1)   > '-1.1'::numeric
ok 106 - check casted vs uncasted for '0.1'::numeric(2,1)   > '0.1'::numeric
ok 107 - check casted vs uncasted for '0.1'::numeric(2,1)   > '1.1'::numeric
ok 108 - check casted vs uncasted for '0.1'::numeric(2,1)   > NULL::numeric
ok 109 - check casted vs uncasted for '0.1'::numeric(2,1)   > '-1.1'::numeric(2,1)
ok 110 - check casted vs uncasted for '0.1'::numeric(2,1)   > '0.1'::numeric(2,1)
ok 111 - check casted vs uncasted for '0.1'::numeric(2,1)   > '1.1'::numeric(2,1)
ok 112 - check casted vs uncasted for '0.1'::numeric(2,1)   > NULL::numeric(2,1)
ok 113 - check casted vs uncasted for '0.1'::numeric !=  '-1.1'::numeric
ok 114 - check casted vs uncasted for '0.1'::numeric !=  '0.1'::numeric
ok 115 - check casted vs uncasted for '0.1'::numeric !=  '1.1'::numeric
ok 116 - check casted vs uncasted for '0.1'::numeric !=  NULL::numeric
ok 117 - check casted vs uncasted for '0.1'::numeric !=  '-1.1'::numeric(2,1)
ok 118 - check casted vs uncasted for '0.1'::numeric !=  '0.1'::numeric(2,1)
ok 119 - check casted vs uncasted for '0.1'::numeric !=  '1.1'::numeric(2,1)
ok 120 - check casted vs uncasted for '0.1'::numeric !=  NULL::numeric(2,1)
ok 121 - check casted vs uncasted for '0.1'::numeric(2,1) !=  '-1.1'::numeric
ok 122 - check casted vs uncasted for '0.1'::numeric(2,1) !=  '0.1'::numeric
ok 123 - check casted vs uncasted for '0.1'::numeric(2,1) !=  '1.1'::numeric
ok 124 - check casted vs uncasted for '0.1'::numeric(2,1) !=  NULL::numeric
ok 125 - check casted vs uncasted for '0.1'::numeric(2,1) !=  '-1.1'::numeric(2,1)
ok 126 - check casted vs uncasted for '0.1'::numeric(2,1) !=  '0.1'::numeric(2,1)
ok 127 - check casted vs uncasted for '0.1'::numeric(2,1) !=  '1.1'::numeric(2,1)
ok 128 - check casted vs uncasted for '0.1'::numeric(2,1) !=  NULL::numeric(2,1)
ok 129 - check variant vs uncasted for '0.1'::numeric <   '-1.1'::numeric
ok 130 - check variant vs uncasted for '0.1'::numeric <   '0.1'::numeric
ok 131 - check variant vs uncasted for '0.1'::numeric <   '1.1'::numeric
ok 132 - check variant vs uncasted for '0.1'::numeric <   NULL::numeric
ok 133 - check variant vs uncasted for '0.1'::numeric <   '-1.1'::numeric(2,1)
ok 134 - check variant vs uncasted for '0.1'::numeric <   '0.1'::numeric(2,1)
ok 135 - check variant vs uncasted for '0.1'::numeric <   '1.1'::numeric(2,1)
ok 136 - check variant vs uncasted for '0.1'::numeric <   NULL::numeric(2,1)
ok 137 - check variant vs uncasted for '0.1'::numeric(2,1) <   '-1.1'::numeric
ok 138 - check variant vs uncasted for '0.1'::numeric(2,1) <   '0.1'::numeric
ok 139 - check variant vs uncasted for '0.1'::numeric(2,1) <   '1.1'::numeric
ok 140 - check variant vs uncasted for '0.1'::numeric(2,1) <   NULL::numeric
ok 141 - check variant vs uncasted for '0.1'::numeric(2,1) <   '-1.1'::numeric(2,1)
ok 142 - check variant vs uncasted for '0.1'::numeric(2,1) <   '0.1'::numeric(2,1)
ok 143 - check variant vs uncasted for '0.1'::numeric(2,1) <   '1.1'::numeric(2,1)
ok 144 - check variant vs uncasted for '0.1'::numeric(2,1) <   NULL::numeric(2,1)
ok 145 - check variant vs uncasted for '0.1'::numeric <=  '-1.1'::numeric
ok 146 - check variant vs uncasted for '0.1'::numeric <=  '0.1'::numeric
ok 147 - check variant vs uncasted for '0.1'::numeric <=  '1.1'::numeric
ok 148 - check variant vs uncasted for '0.1'::numeric <=  NULL::numeric
ok 149 - check variant vs uncasted for '0.1'::numeric <=  '-1.1'::numeric(2,1)
ok 150 - check variant vs uncasted for '0.1'::numeric <=  '0.1'::numeric(2,1)
ok 151 - check variant vs uncasted for '0.1'::numeric <=  '1.1'::numeric(2,1)
ok 152 - check variant vs uncasted for '0.1'::numeric <=  NULL::numeric(2,1)
ok 153 - check variant vs uncasted for '0.1'::numeric(2,1) <=  '-1.1'::numeric
ok 154 - check variant vs uncasted for '0.1'::numeric(2,1) <=  '0.1'::numeric
ok 155 - check variant vs uncasted for '0.1'::numeric(2,1) <=  '1.1'::numeric
ok 156 - check variant vs uncasted for '0.1'::numeric(2,1) <=  NULL::numeric
ok 157 - check variant vs uncasted for '0.1'::numeric(2,1) <=  '-1.1'::numeric(2,1)
ok 158 - check variant vs uncasted for '0.1'::numeric(2,1) <=  '0.1'::numeric(2,1)
ok 159 - check variant vs uncasted for '0.1'::numeric(2,1) <=  '1.1'::numeric(2,1)
ok 160 - check variant vs uncasted for '0.1'::numeric(2,1) <=  NULL::numeric(2,1)
ok 161 - check variant vs uncasted for '0.1'::numeric  =  '-1.1'::numeric
ok 162 - check variant vs uncasted for '0.1'::numeric  =  '0.1'::numeric
ok 163 - check variant vs uncasted for '0.1'::numeric  =  '1.1'::numeric
ok 164 - check variant vs uncasted for '0.1'::numeric  =  NULL::numeric
ok 165 - check variant vs uncasted for '0.1'::numeric  =  '-1.1'::numeric(2,1)
ok 166 - check variant vs uncasted for '0.1'::numeric  =  '0.1'::numeric(2,1)
ok 167 - check variant vs uncasted for '0.1'::numeric  =  '1.1'::numeric(2,1)
ok 168 - check variant vs uncasted for '0.1'::numeric  =  NULL::numeric(2,1)
ok 169 - check variant vs uncasted for '0.1'::numeric(2,1)  =  '-1.1'::numeric
ok 170 - check variant vs uncasted for '0.1'::numeric(2,1)  =  '0.1'::numeric
ok 171 - check variant vs uncasted for '0.1'::numeric(2,1)  =  '1.1'::numeric
ok 172 - check variant vs uncasted for '0.1'::numeric(2,1)  =  NULL::numeric
ok 173 - check variant vs uncasted for '0.1'::numeric(2,1)  =  '-1.1'::numeric(2,1)
ok 174 - check variant vs uncasted for '0.1'::numeric(2,1)  =  '0.1'::numeric(2,1)
ok 175 - check variant vs uncasted for '0.1'::numeric(2,1)  =  '1.1'::numeric(2,1)
ok 176 - check variant vs uncasted for '0.1'::numeric(2,1)  =  NULL::numeric(2,1)
ok 177 - check variant vs uncasted for '0.1'::numeric  >= '-1.1'::numeric
ok 178 - check variant vs uncasted for '0.1'::numeric  >= '0.1'::numeric
ok 179 - check variant vs uncasted for '0.1'::numeric  >= '1.1'::numeric
ok 180 - check variant vs uncasted for '0.1'::numeric  >= NULL::numeric
ok 181 - check variant vs uncasted for '0.1'::numeric  >= '-1.1'::numeric(2,1)
ok 182 - check variant vs uncasted for '0.1'::numeric  >= '0.1'::numeric(2,1)
ok 183 - check variant vs uncasted for '0.1'::numeric  >= '1.1'::numeric(2,1)
ok 184 - check variant vs uncasted for '0.1'::numeric  >= NULL::numeric(2,1)
ok 185 - check variant vs uncasted for '0.1'::numeric(2,1)  >= '-1.1'::numeric
ok 186 - check variant vs uncasted for '0.1'::numeric(2,1)  >= '0.1'::numeric
ok 187 - check variant vs uncasted for '0.1'::numeric(2,1)  >= '1.1'::numeric
ok 188 - check variant vs uncasted for '0.1'::numeric(2,1)  >= NULL::numeric
ok 189 - check variant vs uncasted for '0.1'::numeric(2,1)  >= '-1.1'::numeric(2,1)
ok 190 - check variant vs uncasted for '0.1'::numeric(2,1)  >= '0.1'::numeric(2,1)
ok 191 - check variant vs uncasted for '0.1'::numeric(2,1)  >= '1.1'::numeric(2,1)
ok 192 - check variant vs uncasted for '0.1'::numeric(2,1)  >= NULL::numeric(2,1)
ok 193 - check variant vs uncasted for '0.1'::numeric   > '-1.1'::numeric
ok 194 - check variant vs uncasted for '0.1'::numeric   > '0.1'::numeric
ok 195 - check variant vs uncasted for '0.1'::numeric   > '1.1'::numeric
ok 196 - check variant vs uncasted for '0.1'::numeric   > NULL::numeric
ok 197 - check variant vs uncasted for '0.1'::numeric   > '-1.1'::numeric(2,1)
ok 198 - check variant vs uncasted for '0.1'::numeric   > '0.1'::numeric(2,1)
ok 199 - check variant vs uncasted for '0.1'::numeric   > '1.1'::numeric(2,1)
ok 200 - check variant vs uncasted for '0.1'::numeric   > NULL::numeric(2,1)
ok 201 - check variant vs uncasted for '0.1'::numeric(2,1)   > '-1.1'::numeric
ok 202 - check variant vs uncasted for '0.1'::numeric(2,1)   > '0.1'::numeric
ok 203 - check variant vs uncasted for '0.1'::numeric(2,1)   > '1.1'::numeric
ok 204 - check variant vs uncasted for '0.1'::numeric(2,1)   > NULL::numeric
ok 205 - check variant vs uncasted for '0.1'::numeric(2,1)   > '-1.1'::numeric(2,1)
ok 206 - check variant vs uncasted for '0.1'::numeric(2,1)   > '0.1'::numeric(2,1)
ok 207 - check variant vs uncasted for '0.1'::numeric(2,1)   > '1.1'::numeric(2,1)
ok 208 - check variant vs uncasted for '0.1'::numeric(2,1)   > NULL::numeric(2,1)
ok 209 - check variant vs uncasted for '0.1'::numeric !=  '-1.1'::numeric
ok 210 - check variant vs uncasted for '0.1'::numeric !=  '0.1'::numeric
ok 211 - check variant vs uncasted for '0.1'::numeric !=  '1.1'::numeric
ok 212 - check variant vs uncasted for '0.1'::numeric !=  NULL::numeric
ok 213 - check variant vs uncasted for '0.1'::numeric !=  '-1.1'::numeric(2,1)
ok 214 - check variant vs uncasted for '0.1'::numeric !=  '0.1'::numeric(2,1)
ok 215 - check variant vs uncasted for '0.1'::numeric !=  '1.1'::numeric(2,1)
ok 216 - check variant vs uncasted for '0.1'::numeric !=  NULL::numeric(2,1)
ok 217 - check variant vs uncasted for '0.1'::numeric(2,1) !=  '-1.1'::numeric
ok 218 - check variant vs uncasted for '0.1'::numeric(2,1) !=  '0.1'::numeric
ok 219 - check variant vs uncasted for '0.1'::numeric(2,1) !=  '1.1'::numeric
ok 220 - check variant vs uncasted for '0.1'::numeric(2,1) !=  NULL::numeric
ok 221 - check variant vs uncasted for '0.1'::numeric(2,1) !=  '-1.1'::numeric(2,1)
ok 222 - check variant vs uncasted for '0.1'::numeric(2,1) !=  '0.1'::numeric(2,1)
ok 223 - check variant vs uncasted for '0.1'::numeric(2,1) !=  '1.1'::numeric(2,1)
ok 224 - check variant vs uncasted for '0.1'::numeric(2,1) !=  NULL::numeric(2,1)
ok 225 - check variant vs casted for '''0.1''::numeric' <   '''-1.1''::numeric'
ok 226 - check variant vs casted for '''0.1''::numeric' <   '''0.1''::numeric'
ok 227 - check variant vs casted for '''0.1''::numeric' <   '''1.1''::numeric'
ok 228 - check variant vs casted for '''0.1''::numeric' <   'NULL::numeric'
ok 229 - check variant vs casted for '''0.1''::numeric' <   '''-1.1''::numeric(2,1)'
ok 230 - check variant vs casted for '''0.1''::numeric' <   '''0.1''::numeric(2,1)'
ok 231 - check variant vs casted for '''0.1''::numeric' <   '''1.1''::numeric(2,1)'
ok 232 - check variant vs casted for '''0.1''::numeric' <   'NULL::numeric(2,1)'
ok 233 - check variant vs casted for '''0.1''::numeric(2,1)' <   '''-1.1''::numeric'
ok 234 - check variant vs casted for '''0.1''::numeric(2,1)' <   '''0.1''::numeric'
ok 235 - check variant vs casted for '''0.1''::numeric(2,1)' <   '''1.1''::numeric'
ok 236 - check variant vs casted for '''0.1''::numeric(2,1)' <   'NULL::numeric'
ok 237 - check variant vs casted for '''0.1''::numeric(2,1)' <   '''-1.1''::numeric(2,1)'
ok 238 - check variant vs casted for '''0.1''::numeric(2,1)' <   '''0.1''::numeric(2,1)'
ok 239 - check variant vs casted for '''0.1''::numeric(2,1)' <   '''1.1''::numeric(2,1)'
ok 240 - check variant vs casted for '''0.1''::numeric(2,1)' <   'NULL::numeric(2,1)'
ok 241 - check variant vs casted for '''0.1''::numeric' <=  '''-1.1''::numeric'
ok 242 - check variant vs casted for '''0.1''::numeric' <=  '''0.1''::numeric'
ok 243 - check variant vs casted for '''0.1''::numeric' <=  '''1.1''::numeric'
ok 244 - check variant vs casted for '''0.1''::numeric' <=  'NULL::numeric'
ok 245 - check variant vs casted for '''0.1''::numeric' <=  '''-1.1''::numeric(2,1)'
ok 246 - check variant vs casted for '''0.1''::numeric' <=  '''0.1''::numeric(2,1)'
ok 247 - check variant vs casted for '''0.1''::numeric' <=  '''1.1''::numeric(2,1)'
ok 248 - check variant vs casted for '''0.1''::numeric' <=  'NULL::numeric(2,1)'
ok 249 - check variant vs casted for '''0.1''::numeric(2,1)' <=  '''-1.1''::numeric'
ok 250 - check variant vs casted for '''0.1''::numeric(2,1)' <=  '''0.1''::numeric'
ok 251 - check variant vs casted for '''0.1''::numeric(2,1)' <=  '''1.1''::numeric'
ok 252 - check variant vs casted for '''0.1''::numeric(2,1)' <=  'NULL::numeric'
ok 253 - check variant vs casted for '''0.1''::numeric(2,1)' <=  '''-1.1''::numeric(2,1)'
ok 254 - check variant vs casted for '''0.1''::numeric(2,1)' <=  '''0.1''::numeric(2,1)'
ok 255 - check variant vs casted for '''0.1''::numeric(2,1)' <=  '''1.1''::numeric(2,1)'
ok 256 - check variant vs casted for '''0.1''::numeric(2,1)' <=  'NULL::numeric(2,1)'
ok 257 - check variant vs casted for '''0.1''::numeric'  =  '''-1.1''::numeric'
ok 258 - check variant vs casted for '''0.1''::numeric'  =  '''0.1''::numeric'
ok 259 - check variant vs casted for '''0.1''::numeric'  =  '''1.1''::numeric'
ok 260 - check variant vs casted for '''0.1''::numeric'  =  'NULL::numeric'
ok 261 - check variant vs casted for '''0.1''::numeric'  =  '''-1.1''::numeric(2,1)'
ok 262 - check variant vs casted for '''0.1''::numeric'  =  '''0.1''::numeric(2,1)'
ok 263 - check variant vs casted for '''0.1''::numeric'  =  '''1.1''::numeric(2,1)'
ok 264 - check variant vs casted for '''0.1''::numeric'  =  'NULL::numeric(2,1)'
ok 265 - check variant vs casted for '''0.1''::numeric(2,1)'  =  '''-1.1''::numeric'
ok 266 - check variant vs casted for '''0.1''::numeric(2,1)'  =  '''0.1''::numeric'
ok 267 - check variant vs casted for '''0.1''::numeric(2,1)'  =  '''1.1''::numeric'
ok 268 - check variant vs casted for '''0.1''::numeric(2,1)'  =  'NULL::numeric'
ok 269 - check variant vs casted for '''0.1''::numeric(2,1)'  =  '''-1.1''::numeric(2,1)'
ok 270 - check variant vs casted for '''0.1''::numeric(2,1)'  =  '''0.1''::numeric(2,1)'
ok 271 - check variant vs casted for '''0.1''::numeric(2,1)'  =  '''1.1''::numeric(2,1)'
ok 272 - check variant vs casted for '''0.1''::numeric(2,1)'  =  'NULL::numeric(2,1)'
ok 273 - check variant vs casted for '''0.1''::numeric'  >= '''-1.1''::numeric'
ok 274 - check variant vs casted for '''0.1''::numeric'  >= '''0.1''::numeric'
ok 275 - check variant vs casted for '''0.1''::numeric'  >= '''1.1''::numeric'
ok 276 - check variant vs casted for '''0.1''::numeric'  >= 'NULL::numeric'
ok 277 - check variant vs casted for '''0.1''::numeric'  >= '''-1.1''::numeric(2,1)'
ok 278 - check variant vs casted for '''0.1''::numeric'  >= '''0.1''::numeric(2,1)'
ok 279 - check variant vs casted for '''0.1''::numeric'  >= '''1.1''::numeric(2,1)'
ok 280 - check variant vs casted for '''0.1''::numeric'  >= 'NULL::numeric(2,1)'
ok 281 - check variant vs casted for '''0.1''::numeric(2,1)'  >= '''-1.1''::numeric'
ok 282 - check variant vs casted for '''0.1''::numeric(2,1)'  >= '''0.1''::numeric'
ok 283 - check variant vs casted for '''0.1''::numeric(2,1)'  >= '''1.1''::numeric'
ok 284 - check variant vs casted for '''0.1''::numeric(2,1)'  >= 'NULL::numeric'
ok 285 - check variant vs casted for '''0.1''::numeric(2,1)'  >= '''-1.1''::numeric(2,1)'
ok 286 - check variant vs casted for '''0.1''::numeric(2,1)'  >= '''0.1''::numeric(2,1)'
ok 287 - check variant vs casted for '''0.1''::numeric(2,1)'  >= '''1.1''::numeric(2,1)'
ok 288 - check variant vs casted for '''0.1''::numeric(2,1)'  >= 'NULL::numeric(2,1)'
ok 289 - check variant vs casted for '''0.1''::numeric'   > '''-1.1''::numeric'
ok 290 - check variant vs casted for '''0.1''::numeric'   > '''0.1''::numeric'
ok 291 - check variant vs casted for '''0.1''::numeric'   > '''1.1''::numeric'
ok 292 - check variant vs casted for '''0.1''::numeric'   > 'NULL::numeric'
ok 293 - check variant vs casted for '''0.1''::numeric'   > '''-1.1''::numeric(2,1)'
ok 294 - check variant vs casted for '''0.1''::numeric'   > '''0.1''::numeric(2,1)'
ok 295 - check variant vs casted for '''0.1''::numeric'   > '''1.1''::numeric(2,1)'
ok 296 - check variant vs casted for '''0.1''::numeric'   > 'NULL::numeric(2,1)'
ok 297 - check variant vs casted for '''0.1''::numeric(2,1)'   > '''-1.1''::numeric'
ok 298 - check variant vs casted for '''0.1''::numeric(2,1)'   > '''0.1''::numeric'
ok 299 - check variant vs casted for '''0.1''::numeric(2,1)'   > '''1.1''::numeric'
ok 300 - check variant vs casted for '''0.1''::numeric(2,1)'   > 'NULL::numeric'
ok 301 - check variant vs casted for '''0.1''::numeric(2,1)'   > '''-1.1''::numeric(2,1)'
ok 302 - check variant vs casted for '''0.1''::numeric(2,1)'   > '''0.1''::numeric(2,1)'
ok 303 - check variant vs casted for '''0.1''::numeric(2,1)'   > '''1.1''::numeric(2,1)'
ok 304 - check variant vs casted for '''0.1''::numeric(2,1)'   > 'NULL::numeric(2,1)'
ok 305 - check variant vs casted for '''0.1''::numeric' !=  '''-1.1''::numeric'
ok 306 - check variant vs casted for '''0.1''::numeric' !=  '''0.1''::numeric'
ok 307 - check variant vs casted for '''0.1''::numeric' !=  '''1.1''::numeric'
ok 308 - check variant vs casted for '''0.1''::numeric' !=  'NULL::numeric'
ok 309 - check variant vs casted for '''0.1''::numeric' !=  '''-1.1''::numeric(2,1)'
ok 310 - check variant vs casted for '''0.1''::numeric' !=  '''0.1''::numeric(2,1)'
ok 311 - check variant vs casted for '''0.1''::numeric' !=  '''1.1''::numeric(2,1)'
ok 312 - check variant vs casted for '''0.1''::numeric' !=  'NULL::numeric(2,1)'
ok 313 - check variant vs casted for '''0.1''::numeric(2,1)' !=  '''-1.1''::numeric'
ok 314 - check variant vs casted for '''0.1''::numeric(2,1)' !=  '''0.1''::numeric'
ok 315 - check variant vs casted for '''0.1''::numeric(2,1)' !=  '''1.1''::numeric'
ok 316 - check variant vs casted for '''0.1''::numeric(2,1)' !=  'NULL::numeric'
ok 317 - check variant vs casted for '''0.1''::numeric(2,1)' !=  '''-1.1''::numeric(2,1)'
ok 318 - check variant vs casted for '''0.1''::numeric(2,1)' !=  '''0.1''::numeric(2,1)'
ok 319 - check variant vs casted for '''0.1''::numeric(2,1)' !=  '''1.1''::numeric(2,1)'
ok 320 - check variant vs casted for '''0.1''::numeric(2,1)' !=  'NULL::numeric(2,1)'
